### PR TITLE
Fixes #16

### LINKED ISSUE #16: Incorrectly ignores first word of Authorization header rather than verifying it's "Negotiate"
In the following line, WSGI-Kerberos incorrectly ignores the first word of the `Authorization` header, rather than verifying that it's "Negotiate" before attempting the `self._authenticate` call:
https://github.com/mkomitee/wsgi-kerberos/blob/98e1e745dbae918b8d3af143c835e4dc32d94de0/wsgi_kerberos.py#L187

Instead, WSGI-Kerberos should check if the first word is "Negotiate", and if not, call `self._unauthorized()`.

### PR DESCRIPTION
**Pull request by twosigmajab**
Verified that the tests pass under Python 2 and 3 with these changes.

### CODE REVIEW

File: wsgi_kerberos.py [hunk 1]
```diff
@@ -182,9 +182,17 @@ def __call__(self, environ, start_response):
...
-        # attempt to authenticate with it.
-        client_token = ''.join(authorization.split()[1:])
+        # We have an 'Authorization' header -> should start with 'Negotiate '.
+        parsed = authorization.split(None, 1)
+        if len(parsed) < 2 or parsed[0].lower() != 'negotiate':
+            return self._unauthorized(
+                environ,
+                start_response,
+                # If extra_body is added from PR #10, use it to explain what went wrong:
+                # extra_body=b': Expected "Authorization: Negotiate <token>"',
```
Comment (propertone) [inline wsgi_kerberos.py]: Remove this

**Review by propertone — CHANGES_REQUESTED** [review]

File: wsgi_kerberos.py (hunk 1 above)
Comment (twosigmajab) [reply wsgi_kerberos.py]: Done in the latest revision. Another look?

Comment (twosigmajab) [reply wsgi_kerberos.py]: Note, I updated #18 to incorporate this change. If this PR now looks good to merge, and #18 does too, you can just merge #18 and it should still automatically resolve this PR (#17) as well as issue #16.

**Review by propertone — APPROVED** [review]
Looks good. I'll review merging this as part of #18.

**Comment by propertone** [comment]
Fixed in #18

**Pull request closed without merging**